api/folders.js

```diff
@@ -0,0 +1,52 @@
+var fs = require('fs'),
+    path = require('path'),
+    mkdirp = require('mkdirp'),
+    _ = require('underscore'),
+    config = require('config'),
+    MQ = require('../message-queue');
+
+function Folders() {
+    this.base = './data';
+    
+    this.path = 'null';
+}
+
+Folders.prototype = {
+    init: function() {
+        MQ.publish('main:folder', function(folder) {
+            this.path = path.join(folder, this.base);
```

Comment:
I noticed that path.join is frequently used in this project. Arguments "must be strings" otherwise "an error is thrown": https://nodejs.org/api/path.html#path_path_join_path1_path2

Shall we create our own utility method with appropriate type checks and make it a single point of entry?